compiler/modules/nback.llvm/llvm.bal

```diff
@@ -90,6 +91,16 @@ public class Module {
         return fn;
     }
 
+    public function addFunctionDecl(string name, FunctionType fnType) returns FunctionDecl{
+        FunctionDecl fn = new(name, fnType);
+        self.externalFunctions.push(fn);
+        return fn;
+    }
+
+    public function addExternalFunction(FunctionDefn fn) returns FunctionDecl {
```

Comment:
Why do we need this?

Reply:
My idea was that if you have already define the function in a different module (using our API) you can directly pass the function definition object to this module and it will take care of adding it correctly.

Reply:
Let's cross that bridge when we come to it. We need to look at how LLVM deals with multiple modules. Please leave this out for now.

Reply:
Fixed